bids/variables/kollekshuns.py

```diff
@@ -319,16 +319,18 @@ def resample(self, sampling_rate=None, variables=None, force_dense=False,
                 continue
             if isinstance(var, SparseRunVariable):
                 if force_dense and is_numeric_dtype(var.values):
-                    _variables[name] = var.to_dense(sampling_rate)
-            else:
-                # None if in_place; no update needed
-                _var = var.resample(sampling_rate,
-                                    inplace=in_place,
-                                    kind=kind)
-                if not in_place:
-                    _variables[name] = _var
-
-        if in_place:
+                    var = var.to_dense(self.sampling_rate)
```

Comment:
This puts a lot of trust in the collection sampling rate to be high enough to resolve any given sparse variable well enough. As discussed, elsewhere, there's a "natural" sampling rate that comes with onset/durations, which is the (reciprocal of the) smallest unit that divides evenly into the onset and the duration. What if we updated `to_dense()` so that the default behavior is to use the natural density, then called this without an argument?

Reply:
The main downside I see to this approach is that unless you do this at collection level, you'll end up with a different sampling rate for each variable, which means if you have any transformations that involve multiple dense variables (e.g., orthogonalization), you'll then end up resampling unnecessarily a bunch of times. If you have a sane default that applies to all variables, you avoid that.

The other approach would be to base the sampling rate for all variables on the highest sampling rate detected for any one of the variables, but (a) this requires access to a collection (doesn't work if called directly on a `BIDSVariable`, and more importantly (b) it slows things down because now a bunch of variables are unnecessarily oversampled.

I don't really have a strong preference between any of these options, and in a sense it's an empirical question. In an ideal world, we'd run an extensive set of simulations and see how much  distortion we get in the final regressors under realistic assumptions (well, I guess in an ideal world, we'd just do everything symbolically). But I don't think it's worth the time investment, so I'll defer to your judgment on this. If you want to go with this approach, it's fine with me.

Reply:
@tyarkoni In the context of `BIDSRunVariableCollection.resample()`, this is just temporarily resampling to the "natural" rate, but the output sampling rate of the operation hasn't changed. This change is already switching from directly densifying to the target sampling rate to some intermediate (typically higher) rate before resampling to the target and taking advantage of @satra's anti-aliasing improvements.

Reply:
@effigies Just talked to Tal. We decided this works as is, as it solves our problem.

But ideally, we would want to the `Transformations` module to also use your new upscaling logic, whereas currently it uses the fixed collection rate of 10hz. For example, if you apply `Orthogonalize` to a Sparse variable, it'll use 10hz.

The downside of having each variable use their own sampling_rate is that it increases the chances of multiple resamplings down the line. 

For now, if users want to risk that, and use the "smart" `sampling_rate`, they can add a `ToDense` transformation manually. We can decide later if the default should be this later.

Reply:
@effigies yeah, I just chatted with @adelavega about it. I think we're on the same page with respect to the immediate need for this. My only residual concern is whether we should also change the default behavior of the `ToDense` transformation. But since your PR doesn't change the existing behavior, we can take that up separately.